src/vanilla/store.ts

```diff
@@ -594,6 +583,17 @@ export const createStore = () => {
     if (import.meta.env?.MODE !== 'production') {
       mountedAtoms.add(atom)
     }
+    // mount dependencies before mounting self
```

Comment:
I wonder if we should mount dependencies before mounting self or after mounting self. There was a failing test in #2029 AFAIR, but hmm maybe it's fixed afterwards.